packages/devtools_app/lib/src/ui/search.dart

```diff
@@ -899,3 +960,17 @@ mixin TreeDataSearchStateMixin<T extends TreeNode<T>>
 
 class AutoCompleteController extends DisposableController
     with SearchControllerMixin, AutoCompleteSearchControllerMixin {}
+
+class AutoCompleteMatchSegment {
```

Comment:
instead of creating an AutoComplete specific class for this, can you use Range from utils.dart?

Reply:
Oh nice, done